_posts/2025-11-19-introducing-agentic-search.md

```diff
@@ -0,0 +1,261 @@
+---
+layout: post
+title:  "Introducing agentic search in OpenSearch - Transforming data interaction through natural language"
+authors:
+  - kazabdu
+  - rithinp
+  - ohltyler
+  - jpalis
+  - agtunnell
+  - mingshl
+  - seanzheng
+  - kolchfa
+
+date: 2025-11-19
+has_science_table: true
+categories:
+  - technical-posts
+meta_keywords: agents, tools, agentic, search, LLM, NLQ
+meta_description: Discover how agentic search in OpenSearch 3.3 lets you find data using natural language, without writing complex queries or learning technical syntax.
+---
+
+With OpenSearch 3.3, we are excited to introduce _agentic search_, a major evolution in the way you interact with your data. Agentic search lets you explore your data using a natural language interface, without needing to construct DSL queries manually.
+
+## Agentic search core concepts
+
+Agentic search operates using two key components: agents and tools.
+
+### Agents
+
+An _agent_ is a system powered by a large language model (LLM) that can interpret your request, plan the necessary steps, and decide which tools to use. It understands intent, breaks tasks into actions, and adapts based on intermediate results.
+
+### Tools
+
+_Tools_ are capabilities the agent can invoke, such as listing indexes or performing search. The agent selects the tools to use based on the query intent. For a full list of built-in tools, see [Tools](https://docs.opensearch.org/latest/ml-commons-plugin/agents-tools/tools/index/).
+
+## How agentic search works
+
+Agentic search is powered by an intelligent, agent-driven system that interprets your natural language queries, selects the appropriate tools, and generates optimized search strategies automatically. You can run agentic search using the `agentic` query type, allowing you to explore your data without needing to know query syntax or internal data structures.
+
+For example, you can ask questions such as:
+
+* "Find red cars under $30,000"
+* "Show me last quarter's sales trends"
+* "What are the top performing products in the electronics category?"
+
+The agent then carries out the necessary steps—such as identifying relevant indexes, planning the query, or gathering supplemental information—and returns results aligned with your intent. It also provides a detailed explanation of the tools and decisions involved.
```

Comment:
```suggestion
The agent then carries out the necessary steps, such as identifying relevant indexes, planning the query, or gathering supplemental information, and returns results aligned with your intent. It also provides a detailed explanation of the tools and decisions involved.
```

Reply:
This can be done with either em dashes or commas. cc: @natebower

Reply:
I will leave that to you